scout/parse/clinvar.py

```diff
@@ -0,0 +1,216 @@
+    """
+
+    clinvar_header  = list(CLIVAR_HEADER)
+    optional_fields = dict(CLINVAR_OPTIONAL)
+    clinvars_dict = {}
+
+    if 'all_vars' in form_fields:
+        clinvars = form_fields['local'] #this is a list
+    else:
+        clinvars = [form_fields['main_var'][0]] #also a list, but has one element
+
+    for clinvar in clinvars:
+        clinvar_dict = {}
+        for field in clinvar_header:
+            clinvar_dict[field] = ''
+            if field == 'Gene symbol': # field could contain multiple values
+                clinvar_dict[field] = ';'.join(form_fields[field+'_'+clinvar])
+            elif field == 'Reference sequence' or field == 'HGVS':
+                if 'Reference sequence_'+clinvar in form_fields: #value is provided on the form
+                    ref_HGVS = form_fields['Reference sequence_'+clinvar] # this is a list
+                    if field == 'Reference sequence':
+                        refseq = ref_HGVS[0].split('|')
+                        clinvar_dict[field] = refseq[0]
+                    else:
+                        hgvs = ref_HGVS[0].split('|')
+                        clinvar_dict[field] = hgvs[1]
+                    optional_fields[field] = True
+                else:
+                    if optional_fields[field] == True:
+                        clinvar_dict[field] = ''
+            else: #optional fields
+                if field+"_"+clinvar in form_fields: # optional field is provided
+                    field_value = form_fields[field+'_'+clinvar][0]
+                    if len(field_value) > 0 and field_value != '-': #it's filled in field
+                        clinvar_dict[field] = field_value
+                        if field in optional_fields: # if it's an optional field but it is provided:
+                            optional_fields[field] = True
+                    else: #not filled in, but was filled in for any other variant in this submission
+                        if field in optional_fields and optional_fields[field] == True:
+                            clinvar_dict[field] = ''
+                else: #not provided, but was provided for another variant in this submission form
+                    if field in optional_fields and optional_fields[field] == True:
+                        clinvar_dict[field] = ''
+        # Add variant line to the dictionary of variants to submit
+        clinvars_dict[clinvar] = clinvar_dict
+
+    # remove empty fields from header:
+    for field, used in optional_fields.items():
+        if used == False: #field is not used, remove it from header:
+            clinvar_header.remove(field)
+
+    variants = [] #one line for each variant
+    for clinvar in clinvars:
+        variant_line = []
+        clinvar_dict = clinvars_dict[clinvar]
+        for field in clinvar_header:
+            variant_line.append('"'+clinvar_dict[field]+'"')
+        variants.append(variant_line)
+
+    #returning a list and a list of lists
```

Comment:
Try to return one thing from each function. If you have to return multiple things, pack them into a dictionary like 
```python
variant_lines = {
    'headers': clinvar_headers,
    'variants': variants
}
return variant_lines
```

Reply:
But I don't need a dictionary, I'm using them as a list and a list of lists immediately after when generating the CSV files and in several other places in the code. Otherwise I would need to extract them from the object all the time..